js/plugins/navigate/index.js

```diff
@@ -45,7 +45,18 @@ export default function (Alpine) {
             })
 
             whenItIsReleased(() => {
-                navigateTo(destination)
+                const cancelableEvent = new CustomEvent('alpine:before-navigate', {
```

Comment:
Can this be cleaned up into a separate method?

Reply:
yes in a min.